contrib/database/sql/option.go

```diff
@@ -88,6 +89,16 @@ func WithChildSpansOnly() Option {
 	}
 }
 
+// WithCustomTag will attach the value to the span tagged by the key.
+func WithCustomTag(key string, value interface{}) Option {
```

Comment:
We don't need any func here? Also, can't people use https://pkg.go.dev/gopkg.in/DataDog/dd-trace-go.v1/contrib/database/sql#WithSpanTags ? It's confusing to have two ways to do it.

Reply:
the usage of `WithSpanTags` is different. take a look at the function signature, the semantic is not consistent with the rest of the code base.
to me `WithSpanTags` looks like a one-off thing since all other implementation follows the same pattern.

I agree the added function achieve a similar goal compared to `WithSpanTags` but I think `WithSpanTags` should be departed in favor of this new method.

Reply:
the implementation of `WithSpanTags` is not identical but equivalent to what I provided here.

for `WithSpanTags`:
 - tags are added to `context` under a specific key
 - when span is created via `StartSpanFromContext`, this key is read from context and tags are attached with `span.SetTag`
 - if one has multiple trace calls with the same tags, you only need to call `WithSpanTags` once if all the traces share the same context / parent context object.

for what I added:
 - tags are added as `StartSpanOption`, passed to `StartSpanFromContext` (which is equivalent of calling `StartSpanFromContext` + `span.SetTag`)
 - the multiple trace call with the same tags: need to specify `WithCustomTag` for every call. 

I'm not too concerned about the overhead by `WithCustomTag` since:
 - context may not be the same for different use cases, where the reusability of `WithSpanTags` may not be as applicable
 - additional calls to `WithCustomTag` is relatively cheap. even if the tag map is expensive, there are ways like create the map object ahead of time to make the call cheaper.
 - better design pattern. 

the original PR which introduced `WithSpanTags` were added after the pattern has been established (see #484) and it didn't explain why this is favored over a new option.